TouchFree/Assets/TouchFree/Scripts/CallToInteract/CallToInteractController.cs

```diff
@@ -103,121 +72,109 @@ void OnDisable()
+    void SetupCTI(bool _immediate = false)
     {
         // If the CTI isn't enabled in the config, disable ourselves and exit out.
-        if (!CallToInteractConfig.Config.Enabled)
+        if (!ConfigManager.Config.ctiEnabled)
         {
-            loadedType = CTIType.None;
+            loadedType = CTIType.NONE;
             return;
         }
 
-        if(!_afterCooldown)
+        if(!_immediate)
         {
-            setupCooldown = 0.5f;
+            if (delayedSetupCoroutine == null)
+            {
+                delayedSetupCoroutine = StartCoroutine(DelayedSetup());
+            }
             return;
         }
 
+        HideCTI();
         PrepareCTIAsset();
 
-        if (loadedType == CTIType.Video)
+        if (loadedType == CTIType.VIDEO)
         {
             InitVideoPlayer();
         }
-
-        OnAllHandsExit();
     }
 
-    void Update()
+    Coroutine delayedSetupCoroutine = null;
+    IEnumerator DelayedSetup()
     {
-        UpdateCTI();
-
-        if(setupCooldown > 0)
-        {
-            setupCooldown -= Time.deltaTime;
-
-            if(setupCooldown <= 0)
-            {
-                SetupCTI(true);
-            }
-        }
+        yield return new WaitForSeconds(0.5f);
+        SetupCTI(true);
+        delayedSetupCoroutine = null;
     }
 
-    void UpdateCTI()
+    void UIAcitvated()
     {
-        if (loadedType != CTIType.None)
-        {
-            if (!IsShowing && showTimer.Elapsed.TotalSeconds > ShowTimeAfterNoHandPresent && !ScreenManager.Instance.isActive)
-            {
-                showTimer.Reset();
-                ShowCTI();
-            }
-            else if (IsShowing)
-            {
-                if(hideRequirement == HideRequirement.INTERACTION)
-                {
-                    if(interactionHappened)
-                    {
-                        HideCTI();
-                    }
-                }
-                
-                if (hideRequirement == HideRequirement.PRESENT && hideTimer.Elapsed.TotalSeconds > HideTimeAfterHandPresent)
-                {
-                    hideTimer.Reset();
-                    HideCTI();
-                }
-
-                if(ScreenManager.Instance.isActive)
-                {
-                    HideCTI();
-                    OnAllHandsExit();
-                }
-            }
-        }
-        else if (showTimer.IsRunning)
+        if(isShowing)
         {
-            showTimer.Reset();
+            HideCTI();
         }
     }
 
     void OnHandEnter()
     {
-        showTimer.Reset();
-
-        if (IsShowing && !hideTimer.IsRunning)
+        if (isShowing && ConfigManager.Config.ctiHideTrigger == CtiHideTrigger.PRESENCE)
         {
-            hideTimer.Restart();
+            HideCTI();
         }
     }
 
     void OnAllHandsExit()
     {
-        interactionHappened = false;
-        hideTimer.Reset();
+        if (!isShowing && showAfterHandsLostCoroutine != null)
```

Comment:
This may be more about my knowledge of Coroutines than your code, but don't you want to check the coroutine _is_ null before starting it? Would this not restart it constantly?

Reply:
Good catch!